test-integration/docker-compose.yml

```diff
@@ -0,0 +1,117 @@
+x-common-env-variables: &common-env-variables
+  NETWORK_AUTH_TOKEN: "i_am_very_secret_token"
+  BACALHAU_API_PORT: "1234"
+  MINIO_ROOT_USER: "minioadmin"
+  MINIO_ROOT_PASSWORD: "minioadminpass"
+  AWS_ACCESS_KEY_ID: "minioadmin"
+  AWS_SECRET_ACCESS_KEY: "minioadminpass"
+
+networks:
+  bacalhau-network:
+    driver: bridge
+
+volumes:
+  minio-volume:
+    driver: local
+  registry-volume:
+    driver: local
+
+services:
+  bacalhau-minio-node:
+    image: quay.io/minio/minio
+    container_name: bacalhau-minio-node-container
+    command: server /data --console-address ":9001"
+    volumes:
+      - minio-volume:/data
+    restart: always
+    networks:
+      - bacalhau-network
+    environment: *common-env-variables
+    healthcheck:
+      test: [ "CMD", "curl", "-f", "http://localhost:9000/minio/health/live" ]
+      interval: 1s
+      timeout: 5s
+      retries: 30
+      start_period: 2s
+
+  bacalhau-container-img-registry-node:
+    image: bacalhau-container-img-registry-node-image
+    container_name: bacalhau-container-img-registry-container
+    volumes:
+      - registry-volume:/var/lib/registry
+    restart: always
+    networks:
+      - bacalhau-network
+    environment:
+      REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY: /var/lib/registry
+      REGISTRY_HTTP_ADDR: "0.0.0.0:5000"
+      REGISTRY_HTTP_TLS_CERTIFICATE: "/certs/bacalhau-container-img-registry-node.crt"
+      REGISTRY_HTTP_TLS_KEY: "/certs/bacalhau-container-img-registry-node.key"
+    healthcheck:
+      test: [ "CMD-SHELL", "nc -zv localhost 5000" ]
+      interval: 1s
+      timeout: 5s
+      retries: 30
+      start_period: 2s
+
+  bacalhau-requester-node:
+    image: bacalhau-requester-node-image
+    container_name: bacalhau-requester-node-container
+    networks:
+      - bacalhau-network
+    environment: *common-env-variables
+    depends_on:
+      bacalhau-minio-node:
+        condition: service_healthy
+    privileged: true
+    command:
+      - /bin/bash
+      - -c
+      - |
+        bacalhau config set "orchestrator.auth.token" "$${NETWORK_AUTH_TOKEN}" && bacalhau serve --orchestrator -c api.port=$${BACALHAU_API_PORT}
```

Comment:
This can also be written as:
```bash
bacalhau -c api.port=$${BACALHAU_API_PORT} -c orchestrator.auth.token=$${NETWORK_AUTH_TOKEN} serve --orchestrator
```
but I don't have strong feelings on changing this; sharing for posterity.

Something that might be nice to add later: Provide a config file to bacalhau directly instead of flags, e.g.
```yaml
  command:
    - /bin/bash
    - -c
    - |
      cat <<EOF > config.yaml
      API:
        Port: $${BACALHAU_API_PORT}
      Orchestrator:
          Enabled: true
          Auth:
              Token: "$${NETWORK_AUTH_TOKEN}"
      EOF
      bacalhau --config=config.yaml serve
```
I find this declarative approach to be easier to grok and change later, but again no strong feelings on this.

Reply:
I am all in for declarative approach. I would even argue to minimize imperative commands as much as possible, and make the config the contract between the user and the system.

Though for this PR, I would keep it as is for now, since I would prefer to change the whole of Docker Compose integration to declarative.